src/ck-core/ckarrayoptions.h

```diff
@@ -235,6 +236,24 @@ class CkArrayOptions {
     return *this;
   }
   CkArrayOptions& setStaticInsertion(bool b);
+  bool isStaticInsertion() {
+    if (staticInsertion != -1) return staticInsertion;
+    if (numInitial.dimension == 0) return false;
+    bool empty = false;
+    bool shorts = numInitial.dimension > 3;
+    for (int i = 0; i < numInitial.dimension; i++) {
+      if (shorts && ((short*)numInitial.data())[i] == 0) {
+        empty = true;
+        break;
+      } else if (!shorts && numInitial.data()[i] == 0) {
+        empty = true;
+        break;
```

Comment:
Perhaps I'm missing something, but why can't a statically inserted element have an index value of 0 in any dimension?

Reply:
This is just the default. If there are 0 initial elements, then we assume that the array will be created by dynamic insertion.

Reply:
Aha, I misread that as an array index, but it's the size of the initial elements. Makes sense.